examples/lines_bars_and_markers/stairs_demo.py

```diff
@@ -0,0 +1,60 @@
+"""
+=============
+Histline Demo
+=============
+
+This example demonstrates the use of `~.matplotlib.pyplot.stairs`
+for histogram and histogram-like data visualization and an associated
+underlying `.StepPatch` artist, which is
+a contrained version of `.PathPatch` specified by its bins and edges.
```

Comment:
Not sure what you mean by a "contrained version".  "constrained" doesn't make a lot of sense either.  I'd just say that its a specialized version `.PathPatch` that is specified by its bin values and edges.  This allows convenient access to changing the ``values`` and ``edges`` after the artist is created, ie. in an animation.

Reply:
Overall, though this is a really nice example!

Reply:
Well, constrained meaning restricted or less general, but specialized is a better word.